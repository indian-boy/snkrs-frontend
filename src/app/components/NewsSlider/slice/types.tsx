import { FunctionComponent, SVGProps } from 'react';

export interface SliderItem {
  id: number;
  title: string;
  link: {
    label: string;
    url: string;
  };
  Image: FunctionComponent<
    SVGProps<SVGSVGElement> & { title?: string | undefined }
  >;
}

export interface NewsSliceState {
  selected: SliderItem;
  items: SliderItem[];
}
