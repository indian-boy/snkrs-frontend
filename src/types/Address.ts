export interface Address {
  latitude: number;
  longitude: number;
  location: string;
}
