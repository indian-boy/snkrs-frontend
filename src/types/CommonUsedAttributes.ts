export interface CommonUsedAttributes {
  'data-testid': string;
}
