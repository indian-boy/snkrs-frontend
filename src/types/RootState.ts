import { NewsSliceState } from 'app/components/NewsSlider/slice/types';
import { ThemeState } from 'styles/themes/slice/types';

// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

/*
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
*/
export interface RootState {
  theme?: ThemeState;
  newsSlice?: NewsSliceState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
