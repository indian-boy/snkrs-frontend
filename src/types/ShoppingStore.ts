import { Address } from './Address';

export interface ShoppingStore {
  id: number;
  placeName: string;
  address: Address;
  availability: number;
  serviceHours: string;
}
