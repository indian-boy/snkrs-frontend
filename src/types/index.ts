import { Address } from './Address';
import { RootState } from './RootState';
import { ShoppingStore } from './ShoppingStore';

export type { RootState, Address, ShoppingStore };
