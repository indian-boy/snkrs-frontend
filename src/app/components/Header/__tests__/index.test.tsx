import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { CommonUsedAttributes } from 'types';
import { Header } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const renderWithProviders = (
  props: Parameters<typeof Header>[number] & CommonUsedAttributes,
  store: Store,
) =>
  render(
    <Provider store={store}>
      <Header {...props} />
    </Provider>,
  );

describe('<Header  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = renderWithProviders(
      {
        'data-testid': 'headerID',
      },
      store,
    );

    const headerElement = getByTestId('headerID');

    expect(headerElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
