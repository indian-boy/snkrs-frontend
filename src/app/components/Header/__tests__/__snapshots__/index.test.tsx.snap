// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header  /> should match snapshot 1`] = `
.c5 {
  position: relative;
}

.c5::before {
  position: absolute;
  content: '';
  background-image: url(magnifier.svg);
  background-repeat: no-repeat;
  background-position: center;
  width: 1rem;
  height: 1rem;
  left: 1rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c6 {
  font-size: 1rem;
  padding: 0.9rem 2.5rem 0.9rem 0.5rem;
  outline: none;
  background-color: #FFFFFF;
  border: 1px solid #D6D6D6;
  color: #737373;
  background-color: #F7F7F7;
  border: none;
  border-radius: 3.125rem;
  padding: 0.85rem 2.5rem;
}

.c6::-webkit-search-decoration,
.c6::-webkit-search-cancel-button,
.c6::-webkit-search-results-button,
.c6::-webkit-search-results-decoration {
  display: none;
}

.c6::-webkit-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c6::-moz-placeholder {
  color: #737373;
  opacity: 1;
}

.c6:-ms-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c6::placeholder {
  color: #737373;
  opacity: 1;
}

.c6:-ms-input-placeholder {
  color: #737373;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 1rem;
}

.c2 {
  display: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.875rem;
}

.c1 {
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
}

.c8 {
  width: 1.7rem;
  height: 1.7rem;
  cursor: pointer;
}

.c9 {
  width: 2rem;
  height: 2rem;
  cursor: pointer;
}

.c7 {
  display: none;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000000;
  font-size: 1rem;
  line-height: 1.5rem;
  white-space: nowrap;
  font-weight: bold;
}

.c3:visited {
  color: #1A1A1A;
}

@media (min-width:1024px) {
  .c0 {
    padding: 1rem 2.5rem;
  }
}

@media (min-width:1024px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 1.5rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: 3.75rem;
    height: 3.75rem;
  }
}

@media (min-width:1024px) {
  .c9 {
    display: none;
  }
}

@media (min-width:1024px) {
  .c7 {
    display: block;
  }
}

<div
  class="c0"
>
  <div
    class="styles__LeftSideWrapper-sc-16gafkx-1"
  >
    <svg
      class="c1"
      style="height: 2rem;"
    >
      nike.svg
    </svg>
  </div>
  <div
    class="c2"
  >
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.newOffersPage
      Lançamentos
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.malePage
      Masculino
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.femalePage
      Feminino
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.kidsPage
      Infantil
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.offersPage
      Ofertas
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.newOffersPage
      Lançamentos
    </a>
    <a
      aria-current="page"
      class="c3 active"
      href="/"
    >
      components.header.links.homePage
      SNKRS
    </a>
  </div>
  <div
    class="c4"
  >
    <label
      class="c5"
    >
      <input
        class="c6 c7"
        data-testid="searchID"
        placeholder="components.header.searchButtonLabel,Buscar"
        style="width: 12rem;"
        type="search"
      />
    </label>
    <svg
      class="c8"
    >
      bag.svg
    </svg>
    <svg
      class="c9"
    >
      hamburger_menu.svg
    </svg>
  </div>
</div>
`;
